community/kernel/src/main/java/org/neo4j/kernel/impl/api/cursor/TxAllPropertyCursor.java

```diff
@@ -42,24 +42,29 @@ public TxAllPropertyCursor( Consumer<TxAbstractPropertyCursor> instanceCache )
     @Override
     public boolean next()
     {
+        property = null;
         if ( added == null )
         {
             while ( cursor.next() )
             {
                 int propertyKeyId = cursor.get().propertyKeyId();
 
-                StorageProperty changedProperty = state.getChangedProperty( propertyKeyId );
-                if ( changedProperty != null )
+                if ( state.isPropertyRemoved( propertyKeyId ) )
                 {
-                    this.property = (DefinedProperty) changedProperty;
-                    return true;
+                    continue;
                 }
 
-                if ( !state.isPropertyRemoved( propertyKeyId ) )
+                if ( property == null )
```

Comment:
This code has been removed in later commits, so there is no need to add comments.